DUI3-DX/Sdk/Speckle.Converters.Common/RecursiveConveterResolver.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using Speckle.Autofac.DependencyInjection;
+using Speckle.Converters.Common.Objects;
+
+namespace Speckle.Converters.Common;
+
+public sealed class RecursiveConverterResolver<TConverter> : IConverterResolver<TConverter>
+  where TConverter : class
+{
+  private readonly IFactory<string, TConverter> _factory;
+
+  public RecursiveConverterResolver(IFactory<string, TConverter> factory)
+  {
+    _factory = factory;
+  }
+
+  public TConverter? GetConversionForType(Type objectType)
+  {
+    if (objectType.BaseType == null)
+    {
+      //We've reached the top of the inheritance tree
+      return null;
+    }
+
+    return _factory.ResolveInstance(objectType.Name) ?? GetConversionForType(objectType.BaseType);
```

Comment:
I see this is a layer on top of the factory. Seems fine. Is this pattern in Revit at all yet? I guess this is on the receive side mind

Reply:
This pattern was originally in Revit for ToSpeckle.
This PR moves it to a shared place, and can be used for both send and receive.

Right now, just used in Revit for Send, and Rhino & Autocad for Receive